### PR TITLE
Fix to show the Modules tab in the product editor - global hook fix

### LINKED ISSUE #19246: The modules tab doesn't appear in the product editor
#### Describe the bug

After install a module using the hook "displayAdminProductsExtra" the module doesn't appear in the module tab in the product editor because the modules tab doesn't appear.

#### Expected behavior

The modules tab in the product editor should appear after install the module using the hook "displayAdminProductsExtra".

#### Steps to Reproduce

1. Install a module using the hook "displayAdminProductsExtra"
2. Edit any product
3. The Modules tab doesn't appear.

**Screenshots**

Example without fix

#### Additional information

* PrestaShop version: 1.7.7.0
* PHP version: 7.2.31

### LINKED ISSUE #19439: displayAdminOrderTabLink and displayAdminOrderTabContent not available
#### Describe the bug

We create a test module to show that displayAdminOrderTabLink and displayAdminOrderTabContent  are not available.

image

The tab cannot be displayed.

[This code always return 0](https://github.com/PrestaShop/PrestaShop/blob/6cd77a56c5c7705589185c8ce5709f109c5b54af/src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig#L114) : 
if hookcount('displayAdminOrderTabLink') > 0 or hookcount('displayAdminOrderTabContent') > 0

#### Expected behavior

displayAdminOrderTabLink and displayAdminOrderTabContent must be displayed

So we try tom [comment this line](https://github.com/PrestaShop/PrestaShop/blob/6cd77a56c5c7705589185c8ce5709f109c5b54af/src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/details.html.twig#L114) 

image

#### Steps to Reproduce

1. Install 1.7.7
2. [Install this module](https://github.com/PrestaShop/PrestaShop/files/4695351/tottesthooks.zip) prepared by @jhoarau 
3. See the bug or don't see the tab

Please note I'm focus on these new hooks, but I found that "hookcount" twig function is used for hook displayAdminProductsExtra. So we had this hook on the test module and it looks like "Module tab" on product page is not displayed... I am not sure defining displayAdminProductsExtra
is enought to show Module tab. To be verify...

#### Additional information

* PrestaShop version: 1.7.7
* PHP version: N/A

### PR DESCRIPTION
**Pull request by ghost**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fix to show the Modules tab in the product editor in the Back Office.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #19246
| How to test?  | Install a module using the hook "displayAdminProductsExtra" (@Matt75 has shared a sample module : https://github.com/Matt75/testhookdisplayadminproductsextra), edit any product and the Modules tab doesn't appear. <br>**Example without fix:**<br>Example without fix. <br>**Example with fix:**<br>Example with fix.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19216)

### CODE REVIEW

**Comment by Progi1984** [comment]
@Rolige  Thank you for your contribution. :hugs: 

Could you explain in [an issue your initial bug for this PR](https://github.com/PrestaShop/PrestaShop/issues/new/choose) (or may be link to an issue) ?

Thanks :+1:

**Comment by Rolige** [comment]
> @Rolige Thank you for your contribution. 🤗
> 
> Could you explain in [an issue your initial bug for this PR](https://github.com/PrestaShop/PrestaShop/issues/new/choose) (or may be link to an issue) ?
> 
> Thanks 👍

Really? Do I need to open an issue first before publish a PR?

**Comment by PierreRambaud** [comment]
@Rolige Yes, it's easier to determine if the fix is relevant and need to be merge.
It's a request from @PrestaShop/prestashop-product-team :wink:

**Comment by Rolige** [comment]
> @Rolige Yes, it's easier to determine if the fix is relevant and need to be merge.
> It's a request from @PrestaShop/prestashop-product-team 😉

Issue opened #19216, but this was a wasted time.

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
@PrestaShop/prestashop-product-team I think this fix must be done in the 1.7.7.x wdyt?

**Comment by colinegin** [comment]
Depends if it is a regression, i'm waiting for a better qualification of the issue

**Comment by colinegin** [comment]
I confirm this is not the correct branch, it should target 177x

**Comment by Matt75** [comment]
Can be tested with https://github.com/Matt75/testhookdisplayadminproductsextra

**Comment by Progi1984** [comment]
Thanks @Matt75. I modify the PR ;)

**Comment by matks** [comment]
> but this was a wasted time.

I agree

**Comment by matks** [comment]
So the reason the feature was broken in 1.7.7 is because of PR https://github.com/PrestaShop/PrestaShop/pull/14249 which fixed issue https://github.com/PrestaShop/PrestaShop/issues/10412

TLDR: hooks are lowercased to ease matching, so any logic aiming to scan them, like `hookCount()` must look for lowercase hook names only

**Review by matks — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
It's OK on develop branch, thanks @Rolige !

@matks is going to make the 1.7.7.x PR :wink:

**Comment by Progi1984** [comment]
Thanks @Rolige

**Comment by matks** [comment]
Here it is https://github.com/PrestaShop/PrestaShop/pull/19452
